datumaro/plugins/open_images_format.py

```diff
@@ -214,3 +224,137 @@ def find_sources(cls, path):
+
+    @contextlib.contextmanager
+    def _open_csv_annotation(self, file_name, field_names):
+        absolute_path = osp.join(self._save_dir, OpenImagesPath.ANNOTATIONS_DIR, file_name)
+
+        with open(absolute_path, 'w', encoding='utf-8', newline='') as f:
+            yield csv.DictWriter(f, field_names)
+
+    def apply(self):
+        annotations_dir = osp.join(self._save_dir, OpenImagesPath.ANNOTATIONS_DIR)
+
+        os.makedirs(annotations_dir, exist_ok=True)
+
+        self._save_categories()
+        self._save_label_category_parents()
+        self._save_subsets()
+
+    def _save_categories(self):
+        with self._open_csv_annotation(
+            OpenImagesPath.V5_CLASS_DESCRIPTION_NAME, ['LabelName', 'DisplayName'],
+        ) as class_description_writer:
+            # no .writeheader() here, since we're saving it in the V5 format
+
+            for category in self._extractor.categories()[AnnotationType.label]:
+                class_description_writer.writerow({
+                    'LabelName': category.name,
+                    'DisplayName': category.name,
+                })
+
+    def _save_label_category_parents(self):
+        all_label_names = set()
+        hierarchy_nodes = {}
+        orphan_nodes = []
+
+        def get_node(name):
+            return hierarchy_nodes.setdefault(name, {'LabelName': name})
+
+        for category in self._extractor.categories()[AnnotationType.label]:
+            all_label_names.add(category.name)
+
+            child_node = get_node(category.name)
+
+            if category.parent:
+                parent_node = get_node(category.parent)
+                parent_node.setdefault('Subcategory', []).append(child_node)
+            else:
+                orphan_nodes.append(child_node)
+
+        # The hierarchy has to be rooted in a single node. However, there's
+        # no guarantee that there exists only one orphan (label without a parent).
+        # Therefore, we create a fake root node and make it the parent of every
+        # orphan label.
+        # This is not a violation of the format, because the original OID does
+        # the same thing.
+        root_node = {
+            # Create an OID-like label name that isn't already used by a real label
+            'LabelName': next(root_name
+                for i in itertools.count()
+                for root_name in [f'/m/{i}']
+                if root_name not in all_label_names
+            ),
+            # If an orphan has no children, then it makes no semantic difference
+            # whether it's listed in the hierarchy file or not. So strip such nodes
+            # to avoid recording meaningless data.
+            'Subcategory': [node for node in orphan_nodes if 'Subcategory' in node],
+        }
+
+        hierarchy_path = osp.join(
+            self._save_dir, OpenImagesPath.ANNOTATIONS_DIR, OpenImagesPath.HIERARCHY_NAME)
+
+        with open(hierarchy_path, 'w', encoding='UTF-8') as hierarchy_file:
```

Comment:
```suggestion
        with open(hierarchy_path, 'w', encoding='utf-8') as hierarchy_file:
```
Just for consistency